### PR TITLE
refactor(FX-3669): saved search alert logic

### PR DESCRIPTION
**Pull request by dimatretyak**
The type of this PR is: **Refactor**

This PR resolves [FX-3669]

### PR Checklist (tick all before merging)

- [x] I have included screenshots or videos to illustrate my changes, or I have not changed anything that impacts the UI.
- [x] I have tested my changes on **iOS** and **Android**.
- [x] I have added tests/stories for my changes, or my changes don't require testing/stories, or I have included a link to a separate Jira ticket covering the tests.
- [x] I have added a feature flag, or my changes don't require a feature flag. ([How do I add one?](https://github.com/artsy/eigen/blob/main/docs/developing_a_feature.md))
- [x] I have documented any follow-up work that this PR will require, or it does not require any.
- [x] I have added an app state migration, or my changes do not require one. ([What are migrations?](https://github.com/artsy/eigen/blob/main/docs/adding_state_migrations.md))
- [x] I have added a changelog entry below or my changes do not require one.

### To the reviewers 👀

- [ ] I would like at least one of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- Refactor saved search alert logic - dimatretyak

</details>

[FX-3669]: https://artsyproduct.atlassian.net/browse/FX-3669?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (Saved search alert logic - dimatretyak)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7e66a616d0644317837f636ee31b1e62b6b5c20b
</p>

File: src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx [hunk 1]
```diff
@@ -49,15 +43,19 @@ export const SavedSearchAlertForm: React.FC<SavedSearchAlertFormProps> = (props)
   } = props
   const isUpdateForm = !!savedSearchAlertId
   const isEnabledImprovedAlertsFlow = useFeatureFlag("AREnableImprovedAlertsFlow")
+  const savedSearchPills = SavedSearchStore.useStoreState((state) => state.pills)
+  const attributes = SavedSearchStore.useStoreState((state) => state.attributes)
+  const hasChangedFilters = SavedSearchStore.useStoreState((state) => state.dirty)
+  const removeValueFromAttributesByKeyAction = SavedSearchStore.useStoreActions(
+    (state) => state.removeValueFromAttributesByKeyAction
```
Comment (MounirDhahri) [inline src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx:50]:
```suggestion
    (actions) => actions.removeValueFromAttributesByKeyAction
```

File: src/lib/Scenes/SavedSearchAlert/pillExtractors.ts [hunk 2]
```diff
@@ -58,123 +60,123 @@ export const extractSizeLabel = (prefix: string, value: string) => {
...
+export const extractPills = (attributes: SearchCriteriaAttributes, aggregations: Aggregations): SavedSearchPill[] => {
+  const pills = Object.entries(attributes).map((entry) => {
+    const [paramName, paramValue] = entry as [SearchCriteriaAttributeKeys, any]
 
-    if (isUndefined(paramValue)) {
+    if (isNil(paramValue)) {
       return null
     }
 
-    if (paramName === FilterParamName.width) {
```
Comment (MounirDhahri) [inline src/lib/Scenes/SavedSearchAlert/pillExtractors.ts:135]: **nitpick:** it might be worth it to define a `PillFilterParamName` object maybe 🤷‍♂️

**Review by MounirDhahri — APPROVED** [review]
Thanks Dima, great job as usual. I left 2 small comments but they're not blockers. Feel free to merge

File: src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx (hunk 1 above)
Comment (dimatretyak) [reply src/lib/Scenes/SavedSearchAlert/SavedSearchAlertForm.tsx]: Done

File: src/lib/Scenes/SavedSearchAlert/pillExtractors.ts (hunk 2 above)
Comment (dimatretyak) [reply src/lib/Scenes/SavedSearchAlert/pillExtractors.ts:135]: Done. Made [SearchCriteria](https://github.com/artsy/eigen/blob/87980c3e49f18ca2defa65ee08cd965ff4792793/src/lib/Components/ArtworkFilter/SavedSearch/types.ts#L1-L19) enum

**Comment by MounirDhahri** [comment]
Thanks for the updates Dima 🙇‍♂️ LGTM feel free to merge once George approves

**Review by MounirDhahri — DISMISSED** [review]

**Review by gkartalis — DISMISSED** [review]

**Pull request merged**